exercise/static/exercise/chapter.js

```diff
@@ -579,31 +579,100 @@
           });
         });
       } else {
-        var exercise = this;
-        if (exercise.quiz) {
-          exercise.quizSubmission("submit");
-        } else {
-          /* TODO: migrate to overlay */
-          chapter.openModal(chapter.messages.submit);
+        // Callback for submitting an exercise
+        var submitCallback = function(exercise, hash) {
+          if (exercise.quiz) {
+            exercise.quizSubmission("submit");
+          } else {
+            /* TODO: migrate to overlay */
+            chapter.openModal(chapter.messages.submit);
+          }
+          var url = $(form_element).attr("action");
+          var formData = new FormData(form_element);
+          formData.append("__submission_hash", hash);
+
+          exercise.submitAjax(url, formData, function(data) {
+            //$(form_element).find(":input").prop("disabled", false);
+            //exercise.hideLoader();
+            var input = $(data);
+            if (exercise.quiz) {
+              exercise.quizSubmission("success");
+              exercise.update(input);
+              exercise.renderMath();
+              exercise.dom_element.dispatchEvent(
+                new CustomEvent("aplus:submission-finished",
+                  {bubbles: true, detail: {type: exercise.exercise_type}}));
+            } else {
+              exercise.updateSubmission(input);
+            }
+          });
+        };
+
+        let isDuplicate = false;
```

Comment:
Looks like there is no need at all to define the variable `isDuplicate` outside the function `isDuplicateCallback` that actually uses it. Let's keep the variable's scope as small as it really needed.